interbotix_common_toolbox/interbotix_gravity_compensation/README.md

```diff
@@ -0,0 +1,131 @@
+# interbotix_gravity_compensation
+
+## Overview
+This package contains the C++ implementation of the gravity compensation features of a common robotic arm.
+
+## Gravity Compensation
+### Introduction
+As its name suggests, gravity compensation cancels out the effect of the gravity on a system.
+In our case, we compensate the gravity on a robotic arm with its joint motors.
+This feature is useful when we use the arm as a teleoperation remote or when we teach it a specific trajectory and don't want to keep holding the arm to avoid it from collapsing.
+
+### Formal Formulation
+Computing the torques needed to compensate for the gravity is a special case of the [inverse dynamics](https://en.wikipedia.org/wiki/Inverse_dynamics) problem where the gravity is the only external force.
+More specifically, one is interested in finding the following mapping:
+$$\tau=\text{ID}(\text{model}, \mathbf{q}, \dot{\mathbf{q}}, \ddot{\mathbf{q}})$$
+where $\text{model}$ specifies the inertia of the system and the forces applied to it, $\mathbf{q}$, $\dot{\mathbf{q}}$, and $\ddot{\mathbf{q}}\in\Bbb{R}^6$ are the joint positions, velocities, and accelerations.
+
+One way to solve the inverse dynamics problem is the *recursive Newton-Euler algorithm*.
+It consists three steps:
+1. Compute the velocities and accelerations of the links
+2. Compute the forces required to produce such motions, i.e., fictitious forces
+3. Compute the forces acting upon the links
+
+Step 1: the velocity and acceleration of the $i$'th link $\mathbf{v}_i, \mathbf{a}_i\in\Bbb{S}$ are given by the following recursive relations associated with the angular velocity and acceleration of the $i$'th joint $\dot{q}_i, \ddot{q}_i\in\Bbb{R}$, and the $i$'th joint axis $\mathbf{s}_i\in\Bbb{S}$, where $\Bbb{S}$ is the set of [screws](https://en.wikipedia.org/wiki/Screw_theory).
```

Comment:
Some of this math is not rendering properly. Make sure everything looks good when viewing the file on GitHub.